### PR TITLE
Tear down AppTile using lifecycle tracking

### PR DESCRIPTION
**Pull request by jryans**
This improves confidence that widget frames are torn down correctly by checking whether any other component wants to continue showing it, which is far less error prone than the previous approach of depending on multiple independent stores which can update at different times.

Fixes https://github.com/vector-im/element-web/issues/21025

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Tear down AppTile using lifecycle tracking ([\#7833](https://github.com/matrix-org/matrix-react-sdk/pull/7833)). Fixes vector-im/element-web#21025.

Preview: https://pr7833--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/views/elements/Tooltip.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
```
Comment (t3chguy) [inline src/components/views/elements/Tooltip.tsx:17]: the changes in this file look unrelated?

**Review by t3chguy — APPROVED** [review]
Looks sane!!

File: src/components/views/elements/Tooltip.tsx (hunk 1 above)
Comment (jryans) [reply src/components/views/elements/Tooltip.tsx:17]: Yeah, that's true, though discovered in the course of this work... while working on the test, my editor highlighted that the `this.tooltip` field is never used, so it seemed like a good fix to send along. Do you want it in a separate PR instead?

Comment (t3chguy) [reply src/components/views/elements/Tooltip.tsx:17]: Not at all, just wanted to raise in case it crept in as part of a separate changeset you wanted to commit
